src/tools/openapi.ts

```diff
@@ -92,11 +110,24 @@ Only use the OpenAPI tool if you need to communicate to external servers.`;
                     required: methodSpec.parameters
                       .filter((p: any) => p.required === true)
                       .map((p: any) => p.name),
+                    //properties: methodSpec.parameters.reduce(
+                    //  (acc: any, p: any) => ({
+                    //    ...acc,
+                    //    [p.name]: { ...p.schema, description: p.name },
+                    //  }),
+                    //  {},
+                    //),
                     properties: methodSpec.parameters.reduce(
-                      (acc: any, p: any) => ({
-                        ...acc,
-                        [p.name]: { ...p.schema, description: p.name },
-                      }),
+                      (acc: any, p: any) =>
+                        !p.$ref
+                          ? { ...acc, [p.name]: { ...p.schema, description: p.name } }
+                          : {
+                              ...acc,
+                              [getReferencedObject(this.openApiSchema, p.$ref)?.name]: {
+                                ...getReferencedObject(this.openApiSchema, p.$ref)?.schema,
+                                description: getReferencedObject(this.openApiSchema, p.$ref)?.name,
+                              },
+                            },
```

Comment:
Could you explain and remove comments?

Reply:
I forgot to take out the comments.  The change is for the "$ref" support.
OpenAPI allows to reference to the object defined in the schema.

```
        "parameters": [
          {
            "$ref": "#/components/parameters/org"
          }
        ],
```

```
  "components": {
    "parameters": {
      "org": {
        "name": "org",
        "description": "The organization name. The name is not case sensitive.",
        "in": "path",
        "required": true,
        "schema": {
          "type": "string"
        }
      }
    }
  }
```